### PR TITLE
Select: Preserves focus after options selected

### PR DESCRIPTION
**Pull request by Seebiscuit**
Submitting a PR to add functionality described in #9456. Only implemented in Select in this PR. 

After a user selects an option from the popper menu, the focus reverts back to the `el-input` element "silently". Silently, means that `focus()` is triggered on the input ref (`this.$refs.reference.focus()`), but before that, a bit is set (`this.softFocus` )that disables the popper functionality for this event. 

The way it was implemented is straightforward. After focus fires in `el-input` and reaches `select`, `handleFocus` is invoked. `handleFocus` will test `softFocus`. If `softFocus === true` then `visible` is *not* switched to true, and `softFocus` simply reverts to false. IF the element loses focus after that and gains focus again, `softFocus === false` and the regular functionality of `handleFocus` is invoked, setting `visible = true` and emitting the `focus` event on the element.

### CODE REVIEW

**Comment by Seebiscuit** [comment]
I'm trying to run the tests locally, but I'm getting:

```bash
npm run test

> element-ui@2.2.0 test C:\Development\Libraries\element-ui\element
> npm run lint && npm run build:theme && cross-env CI_ENV=/dev/ karma start test/unit/karma.conf.js --single-run


> element-ui@2.2.0 lint C:\Development\Libraries\element-ui\element
> eslint src/**/* test/**/* packages/**/* build/**/* --quiet


C:\Development\Libraries\element-ui\element\packages\theme-chalk\src\fonts\element-icons.ttf
  1:1  error  Parsing error: Unexpected character ' '

C:\Development\Libraries\element-ui\element\packages\theme-chalk\src\fonts\element-icons.woff
  1:5  error  Parsing error: Unexpected character ' '

✖ 2 problems (2 errors, 0 warnings)
```

Can you please advise?

**Comment by Seebiscuit** [comment]
Also, looking at the Travis CI output, it's not clear to pinpoint which test in the spec failed. How would I go about figuring out what exactly failed?

File: src/index.js [hunk 1]
```diff
@@ -1,142 +1,142 @@
...
-import Collapse from '../packages/collapse/index.js';
-import CollapseItem from '../packages/collapse-item/index.js';
-import Cascader from '../packages/cascader/index.js';
-import ColorPicker from '../packages/color-picker/index.js';
-import Transfer from '../packages/transfer/index.js';
-import Container from '../packages/container/index.js';
-import Header from '../packages/header/index.js';
-import Aside from '../packages/aside/index.js';
-import Main from '../packages/main/index.js';
+import Pagination from '../packages/pagination/index.js';
```
Comment (Leopoldthecoder) [inline src/index.js]: Please revert changes of this file.

File: packages/select/src/select.vue [hunk 2]
```diff
@@ -520,8 +521,14 @@
...
 
       handleFocus(event) {
-        this.visible = true;
-        this.$emit('focus', event);
+        if (!this.softFocus) {
+          console.log('Hit handleFocus will toggle |  softFocus:', this.softFocus);
+          this.visible = true;
+          this.$emit('focus', event);
+        } else {
+          console.log('Hit handleFocus will not toggle |  softFocus:', this.softFocus);
```
Comment (Leopoldthecoder) [inline packages/select/src/select.vue]: Please remove these two `console.log`s.

**Comment by Leopoldthecoder** [comment]
Hi, thanks for contributing!

For local testing, please make sure you have pulled latest code and installed dependencies using `yarn` instead of `npm install`. The error is because eslint is trying to lint font files, which shouldn't happen if your dependencies' versions are correct.

The Travis CI has passed after I rerun it.

Apart from the comments above, please write a test case for this feature (restore focus after selection).

**Comment by Seebiscuit** [comment]
Hey @Leopoldthecoder if this gets merged, how quickly can I pull it from master? I'm trying to build it locally, so I can use it right away, and I'm missing all kinds of dependencies on my main project. Will this be on a upcoming patch or next? Thanks!

**Comment by Leopoldthecoder** [comment]
Hi @Seebiscuit , now there're some conflicts after another PR was merged. Could you resolve them?

We'll release the next version by Sunday, and this PR will be included if everything is alright by then.

**Comment by Seebiscuit** [comment]
Sure. What PR is in conflict, @Leopoldthecoder?  How do I merge that PR into my local working copy to resolve the conflicts?

**Comment by Leopoldthecoder** [comment]
Never mind, I resolved it in the online editor.

**Comment by zhupp034** [comment]
how to disable this focus

**Comment by Seebiscuit** [comment]
@zhupp034 in a coming build, if not the next, select will not automatically open the menu on a tab operation; soft focus will be opt-in using the `automatic-dropdown` attribute.

@leopoldthecoder when is `automatic-dropdown` landing?

**Comment by Leopoldthecoder** [comment]
I think we'll release it this weekend.

**Pull request merged**